src/mosaic-node-manager.cc

```diff
@@ -76,25 +114,71 @@ namespace ns3 {
+            NetDeviceContainer netDevices = m_wifi80211pHelper.Install(m_wifiPhyHelper, m_waveMacHelper, singleNode);
+            m_ipAddressHelper.Assign(netDevices);
+
+            //Install app
+            NS_LOG_INFO("Install MosaicProxyApp application on node " << singleNode->GetId());
+            Ptr<MosaicProxyApp> app = CreateObject<MosaicProxyApp>();
+            app->SetNodeManager(this);
+            singleNode->AddApplication(app);
+            app->SetSockets();
+
+            //Install mobility model
+            NS_LOG_INFO("Install MosaicMobilityModel on node " << singleNode->GetId());
+            Ptr<ConstantVelocityMobilityModel> mobModel = CreateObject<ConstantVelocityMobilityModel>();
+            mobModel->SetPosition(position);
+            singleNode->AggregateObject(mobModel);
+
+        } else if (commType == LTE) {
+
+            // Create and set a mobility model for the node with the given position
+            Ptr<ConstantVelocityMobilityModel> mobModel = CreateObject<ConstantVelocityMobilityModel>();
+            mobModel->SetPosition(position);
+            singleNode->AggregateObject(mobModel);
+            
+            // Associate the node with buildings for better radio propagation modeling
+            BuildingsHelper::Install (singleNode);
+            
+            // Ensure that the mobility models of all nodes are consistent with their positions
+            BuildingsHelper::MakeMobilityModelConsistent(); 
+
+            // Install an LTE device on the node
+            NetDeviceContainer vehDev = m_lteHelper->InstallUeDevice(singleNode);
+
+            // Install the internet stack on the node
+            InternetStackHelper internet;
+            internet.Install(singleNode);
+
+            // Assign an IPv4 address to the LTE device
+            Ipv4InterfaceContainer vehicleIpIface = m_epcHelper->AssignUeIpv4Address(vehDev);
+
+            // Set up static routing for the node to use the default gateway provided by the EPC helper
+            Ipv4StaticRoutingHelper Ipv4RoutingHelper;
+            Ptr<Ipv4StaticRouting> vehicleStaticRouting = Ipv4RoutingHelper.GetStaticRouting(veh->GetObject<Ipv4>());
+            vehicleStaticRouting->SetDefaultRoute(m_epcHelper->GetUeDefaultGatewayAddress(), 1);
+
+            // Attach the LTE device to the eNodeB (base station)
+            m_lteHelper->Attach(vehDev);
+
+            // Create and activate a sidelink bearer for V2X communication
+            Ptr<LteSlTft> tft = Create<LteSlTft>(LteSlTft::TRANSMIT, m_clientRespondersAddress, m_groupL2Address); 
+            m_lteV2xHelper->ActivateSidelinkBearer(Simulator::Now(), vehDev, tft);
+            
+            // Install the V2X sidelink configuration on the LTE device
+            m_lteHelper->InstallSidelinkV2xConfiguration(vehDev, m_ueSidelinkConfiguration);
+
+            // Store the group L2 address and increment it for the next node
+            m_groupL2Addresses.push_back(m_groupL2Address);
```

Comment:
Just to confirm, am I understanding correctly that the names starting with "m_" are member fields of this class, so that this will eventually hold a list of all the initialize LTE device addresses?

Thats what it seems like, and seems correct, but want to double check,

Reply:
I was followed example code to store the group L2 address. However, it seems like this m_groupL2Addresses not being used anymore after I checked back the example code. It now has been removed.